pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -143,8 +154,27 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
-			return ctrl.Result{}, err
+			return r.updateStatus(ctx, req, ms, err)
+		}
+	}
+
+	return r.updateStatus(ctx, req, ms, nil)
+}
+
+func (r *reconciler) updateStatus(ctx context.Context, req ctrl.Request, ms *stack.MonitoringStack, recError error) (ctrl.Result, error) {
+	prom, err := r.monClient.MonitoringV1().Prometheuses(req.Namespace).Get(ctx, ms.Name, metav1.GetOptions{})
+	if err != nil {
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+	}
+	r.conditions.UpdateAvailable(prom.Status.Conditions, ms.Generation)
+	r.conditions.UpdateReconciled(prom.Status.Conditions, ms.Generation, recError)
```

Comment:
Given that `conditions` is short lived and can be always derived from promethues.status.conditions should the reconciler hold onto `conditions` map? This also leads me to think what happens if there are 2 or prometheus instances (CR) wouldn't the condition map get overwritten every time and possibly be incorrect? would that even be a race condition? 



```suggestion
	ms.Status.Conditions = conditionsFromPromethues(prom.Status, ms.Generation, recError)
```

Reply:
Hm good point. You're probably right that it's not safe to hold the conditions map in the controller. Let me try to update it.